src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -410,19 +405,13 @@ public static Instant getBusDepartureTime(Leg busLeg) {
     }
 
     private static double getDistanceToEndOfLeg(TravelerPosition travelerPosition) {
-        return getDistanceToEndOfLeg(travelerPosition, TRIP_INSTRUCTION_UPCOMING_RADIUS);
+        return getDistanceToEndOfLeg(travelerPosition, travelerPosition.getLegPositions());
     }
 
     /**
-     * Get the distance from the traveler's current position to the leg destination.
+     * Get the distance from the traveler's current position to the leg destination from given leg positions.
      */
-    public static double getDistanceToEndOfLeg(TravelerPosition travelerPosition, int radius) {
-        List<Coordinates> legPositions = injectWaypointsIntoLegPositions(
-            travelerPosition.expectedLeg,
-            travelerPosition.expectedLeg.steps,
-            radius
-        );
-
+    public static double getDistanceToEndOfLeg(TravelerPosition travelerPosition, List<Coordinates> legPositions) {
```

Comment:
Given that `legPositions` are cached in `TravelerPostion`, it probably doesn't need passing in separately.

Reply:
This overload is used by a test to cover a case that uses a non-default upcoming threshold. I added Javadoc in https://github.com/ibi-group/otp-middleware/pull/314/commits/5e4489ab3b5b572989b9dc6fb3280ac66fa7935c.